spec/classes/software_vcsscm_git_spec.rb

```diff
@@ -13,6 +13,134 @@
+          it { is_expected.not_to contain_vcsrepo('/opt/bash-git-prompt/') }
+          it { is_expected.not_to contain_file('/etc/bash_completion.d/bash-git-prompt') }
+          it { is_expected.not_to contain_file('/etc/gitconfig') }
+          it { is_expected.not_to contain_file('/etc/skel/.config/git/config') }
+          it { is_expected.not_to contain_file('/etc/skel/.config/git/ignore') }
+        end
+      end
+
+      context 'with gui' do
+        let :params do
+          {
+            :gui => true
+          }
+        end
+        if facts[:operatingsystem] == 'Darwin'
+          it { is_expected.to compile.and_raise_error(/is not supported on Darwin./) }
+        else
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_package('git') }
+          it { is_expected.to contain_package('gitk') }
+          it { is_expected.to contain_package('git-gui') }
+        end
+      end
+
+      context 'with bash completion and prompt' do
+        let :params do
+          {
+            :bash_completion => true,
+            :bash_prompt     => true
+          }
+        end
+        if facts[:operatingsystem] == 'Darwin'
+          it { is_expected.to compile.and_raise_error(/is not supported on Darwin./) }
+        else
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_package('git') }
+          it { is_expected.to contain_package('bash-completion') }
+          it { is_expected.to contain_vcsrepo('/opt/bash-git-prompt/') }
+          it {
+            is_expected.to contain_file('/etc/bash_completion.d/bash-git-prompt')
+              .with_source('puppet:///modules/software/vcsscm/git/bash-git-prompt')
+          }
+        end
+      end
+
+      context 'with booleans for gitconfig and gitignore' do
+        let :params do
+          {
+            :gitconfig => true,
+            :gitignore => true
+          }
+        end
+        if facts[:operatingsystem] == 'Darwin'
+          it { is_expected.to compile.and_raise_error(/is not supported on Darwin./) }
+        else
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_package('git') }
+          it {
+            is_expected.to contain_file('/etc/gitconfig')
+              .with_source('puppet:///modules/software/vcsscm/git/system-gitconfig')
+          }
+          it {
+            is_expected.to contain_file('/etc/skel/.config/git/config')
+              .with_source('puppet:///modules/software/vcsscm/git/user-gitconfig')
+          }
+          it {
+            is_expected.to contain_file('/etc/skel/.config/git/ignore')
+              .with_source('puppet:///modules/software/vcsscm/git/user-gitignore')
+          }
+        end
+      end
+
+      context 'with strings for gitconfig and gitignore' do
+        let :params do
+          {
+            :gitconfig => 'puppet:///modules/software/vcsscm/git/user-gitconfig',
```

Comment:
In the following two examples, could you use some path other than the default here? This way we know it is actually using the users input and not the default values used when set to true...

Reply:
Oh? There test run won't fail because there's no such file? I can try!